### DISCUSSION TITLE
Support applyToActivitiesIfAvailable

### DISCUSSION THREAD

**Discussion by frankois944** [jordond/MaterialKolor#237] [Q&A]
Currently, we need to set seed color.

Is it possible to the equivalent of [DynamicColors.applyToActivitiesIfAvailable](https://developer.android.com/develop/ui/views/theming/dynamic-colors#get-started-views) ?
I know it's only for android, but it can be nice to sync the app color with the OS.

Thanks!

**Comment by faraz152** [comment]
MaterialKolor does not call `DynamicColors.applyToActivitiesIfAvailable()` automatically because it is a KMP library — that API is Android-only and lives in `com.google.android.material`.

However, you can replicate the exact behaviour on Android by reading the system-resolved dynamic color seed and passing it to `DynamicMaterialTheme`. Here are two approaches:

---

**Option A — Read the system accent color (Android 12+ / API 31+)**

```kotlin
@Composable
fun AppTheme(content: @Composable () -> Unit) {
    val context = LocalContext.current

    val seedColor = remember(context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            // System dynamic color — same seed Material uses for its monet palette
            Color(context.getColor(android.R.color.system_accent1_500))
        } else {
            Color(0xFF6750A4.toInt()) // fallback seed for older devices
        }
    }

    DynamicMaterialTheme(
        seedColor = seedColor,
        useDarkTheme = isSystemInDarkTheme(),
        content = content,
    )
}
```

`android.R.color.system_accent1_500` is the wallpaper-derived primary accent — the same value Material's `DynamicColors` uses internally.

---

**Option B — Let Material resolve it, then extract the seed**

If you are already calling `DynamicColors.applyToActivitiesIfAvailable(app)` in your `Application` class for Views-based parts of your app, the dynamic theme is already applied to the Activity context. You can extract the resolved primary:

```kotlin
@Composable
fun AppTheme(content: @Composable () -> Unit) {
    val context = LocalContext.current

    val seedColor = remember(context) {
        val tv = TypedValue()
        context.theme.resolveAttribute(
            com.google.android.material.R.attr.colorPrimary, tv, true
        )
        if (tv.type >= TypedValue.TYPE_FIRST_COLOR_INT) Color(tv.data)
        else Color(0xFF6750A4.toInt())
    }

    DynamicMaterialTheme(
        seedColor = seedColor,
        useDarkTheme = isSystemInDarkTheme(),
        content = content,
    )
}
```

---

On non-Android (Desktop, iOS, Web) targets, just pass a fixed seed or expose it as a parameter — MaterialKolor's theme engine is platform-agnostic so the rest of the color generation works identically everywhere.
